libr/include/r_bin.h

```diff
@@ -233,18 +234,23 @@ typedef struct r_bin_t {
 	bool demanglercmd;
 } RBin;
 
+typedef struct r_bin_xtr_metadata_t {
+	char *arch;
+	int bits;
+	char *libname;
+} RBinXtrMetadata;
+
 typedef int (*FREE_XTR)(void *xtr_obj);
 typedef struct r_bin_xtr_extract_t {
 	char *file;
-	RBuffer *buf;
+	Sdb *sdb;
```

Comment:
I just spotted this. I thought sdb was saved in RBin and shared by all RBinXtrData i don't see the point of have 999 instance of sdb in the case of dyldcache, why  not create only one instance and save sub-binaries there? that way would be even faster and save memory and RBinXtrData describe how to get the data from there not needed sdb in RBinXtrData